### PR TITLE
User story10

### PR DESCRIPTION
**Pull request by BJSherman80**
- [ x ] done

User Story 10, User Registration

As a visitor
When I click on the 'register' link in the nav bar
Then I am on the user registration page ('/register')
And I see a form where I input the following data:
- my name
- my street address
- my city
- my state
- my zip code
- my email address
- my preferred password
- a confirmation field for my password

When I fill in this form completely,
And with a unique email address not already in the system
My details are saved in the database
Then I am logged in as a registered user
I am taken to my profile page ("/profile")
I see a flash message indicating that I am now registered and logged in
and all profile info

### CODE REVIEW

File: app/controllers/welcome_controller.rb [hunk 1]
```diff
@@ -0,0 +1,4 @@
+class WelcomeController<ApplicationController
+def index
```
Comment (helloeduardo) [inline app/controllers/welcome_controller.rb:2]: Will want to indent ➡️

**Review by helloeduardo — APPROVED** [review]
Thanks for getting the sessions worked out! Glad you got it working with `form_tag` instead of `form_for` 🥳

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <%= label_tag "City: " %>
+    <%= text_field_tag :city, @user.city %>
+    <%= label_tag "State: " %>
+    <%= text_field_tag :state, @user.state %>
+    <%= label_tag "Zip Code: " %>
+    <%= text_field_tag :zip, @user.zip %>
+    <%= label_tag "Email Address: " %>
+    <%= text_field_tag :email%>
+    <%= label_tag "Password: " %>
+    <%= text_field_tag :password %>
```
Comment (helloeduardo) [inline app/views/users/new.html.erb:18]: Will we want this text_field to be a password_field eventually? As well as email_field? Not sure if it matters

**Review by nmking22 — CHANGES_REQUESTED** [review]
Looks good, but I think we should add a few more tests:
* Feature testing for show display - we could either do this at the end of our user/new_spec.rb with expectations for the attributes to be displayed or in a new user/show_spec.rb file
* Model testing for password/email validation:
``` 
describe "validations" do
    it {should validate_presence_of(:email)}
    it {should validate_uniqueness_of(:email)}
    it {should validate_presence_of(:password)}
end
```

Also, we may naturally revisit this in later user stories, but we are not currently doing anything with our :password_confirmation. We could put an if statement in our users create action to either create the user as usual if params[:password] == params[:password_confirmation], or redirect back to our users new view with a flash message if they do not match.

**Review by helloeduardo — APPROVED** [review]

File: app/views/users/new.html.erb (hunk 2 above)
Comment (the-color-bliu) [reply app/views/users/new.html.erb:18]: Good idea! I'll second this.

**Review by the-color-bliu — APPROVED** [review]

**Pull request merged**